src/server/svr_movejob.c

```diff
@@ -1360,3 +1370,62 @@ cnvrt_local_move(job *jobp, struct batch_request *req)
 	return (local_move(jobp, req));
 }
 
+/**
+ * @brief
+ * 		small_spool_files - check size of job files
+ * @par
+ * 		Checks the size of the output/error/checkpoint files for a job.
+ * 		If the job is not being rerun, simply returns 1.
+ *
+ * @return	int
+ * @retval	0	: at least one file is larger than 2MB.
+ * @retval	1	: all spool files are smaller than 2MB.
+ */
+static int small_spool_files(job* pjob)
+{
+	int  		max_bytes_over_tpp = 2*1024*1024;
+	char 		path[MAXPATHLEN+1] = {0};
+	struct stat 	sb;
+
+	/* 
+	 * If the job is not being rerun, we need not check
+	 * the size of the spool files.
+	 */
+	if (!(pjob->ji_qs.ji_svrflags & JOB_SVFLG_HASRUN))
+		return 1;
+	
+	(void)strcpy(path, path_spool);
```

Comment:
What I'd rather see happen here is you do your if (*pjob->ji_qs.ji_fileprefix != '\0') statement once and shove the value into a local variable.  Each of your 3 subsequent blocks of strcpy(), strcat(), and strcat() can be turned into snprintf()s.  That is safer than using strcpy() and strcat().